examples/livehunt_network_watch.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/local/bin/python
+# Copyright © 2022 The vt-py authors. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Manage a set of YARA nethunting rules by adding/removing domains.
+
+This script automatically updates a set of nethunting rules (file, url, domain,
+ip) in your VT account by adding and removing domains.
+
+Read more:
+https://www.virustotal.com/gui/hunting-overview
+https://developers.virustotal.com/v3.0/reference#livehunt
+https://developers.virustotal.com/docs/nethunt
+"""
+
+import argparse
+import asyncio
+import copy
+import os
+import re
+import vt
+import sys
+
+
+API_KEY_ENV_VAR = 'VT_API_KEY'
+RULESET_PREFIX = 'auto_network_watch_'
+TEMPLATE_DIR = os.path.join(os.path.dirname(__file__), 'netwatch_templates')
+RULESET_ENTITY = ('file', 'url', 'domain', 'ip_address')
+
+
+def extract_domains_from_rule(rules):
+  """Extract the domain list from the comment of a yara rule."""
+  domain_list = []
+  for line in rules.split('*/')[0].split('---', 2)[1].splitlines():
+    if not line:
+      continue
+    domain = line.split('* ', 2)[1]
+    if domain:
+      domain_list.append(domain)
+  return domain_list
+
+
+async def get_rulesets():
+  """Retrieve a rule from VT to get currently monitored properties."""
+  rulesets = {}
+  async with vt.Client(os.environ.get(API_KEY_ENV_VAR)) as client:
+    try:
+      rulesets_it = client.iterator(
+        '/intelligence/hunting_rulesets',
+        params={'filter': f'name:{RULESET_PREFIX}* tag:autogenerated'},
+        limit=10)
+    except vt.error.APIError as e:
+      print(f'Error retrieving {RULESET_PREFIX}* rulesets: {e}')
+      return rulesets
+
+    async for ruleset in rulesets_it:
+      entity = ruleset.name.split(RULESET_PREFIX)[1]
+      rulesets[entity] = {'id': ruleset.id,
+                        'name': ruleset.name,
+                        'rules': ruleset.rules,
+                        'domains': extract_domains_from_rule(ruleset.rules)}
```

Comment:
will no APIError exceptions be raised while iterating?
https://github.com/VirusTotal/vt-py/blob/023c2e1b8609be997571ff8f83942e3821615bc0/vt/iterator.py#L128-L140

Reply:
done